lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -109,142 +188,175 @@ function getNodeTextLength(node) {
-      const {computedStyle} = result;
-      const fontSizeProperty = computedStyle.find(({name}) => name === FONT_SIZE_PROPERTY_NAME);
-
-      return {
-        fontSize: parseInt(fontSizeProperty.value, 10),
-        textLength: getNodeTextLength(node),
-        node: node.parentNode,
-      };
-    })
-    .catch(err => {
-      require('../../../lib/sentry.js').captureException(err);
-      return null;
+async function fetchComputedFontSize(driver, textNode) {
+  try {
+    const {computedStyle} = await driver.sendCommand('CSS.getComputedStyleForNode', {
+      nodeId: textNode.parentId,
     });
+
+    const fontSizeProperty = computedStyle.find(({name}) => name === FONT_SIZE_PROPERTY_NAME);
+
+    return {
+      // @ts-ignore - font size property guaranteed to be returned in getComputedStyle
+      fontSize: parseInt(fontSizeProperty.value, 10),
+      textLength: getNodeTextLength(textNode),
+      node: textNode.parentNode,
+    };
+  } catch (err) {
+    Sentry.captureException(err, {tags: {gatherer: 'FontSize'}});
+    return null;
+  }
 }
 
 /**
  * @param {LH.Artifacts.FontSize.DomNodeWithParent} node
  * @returns {boolean}
  */
 function isNonEmptyTextNode(node) {
-  return node.nodeType === global.Node.TEXT_NODE &&
+  return (
+    node.nodeType === TEXT_NODE_TYPE &&
     !TEXT_NODE_BLOCK_LIST.has(node.parentNode.nodeName) &&
-    getNodeTextLength(node) > 0;
+    getNodeTextLength(node) > 0
+  );
 }
 
 class FontSize extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext['driver']} driver
+   */
+  static async fetchNodesToAnalyze(driver) {
+    let failingTextLength = 0;
+    let visitedTextLength = 0;
+    let totalTextLength = 0;
+
+    const nodes = await getAllNodesFromBody(driver);
+
+    const textNodes = nodes.filter(isNonEmptyTextNode);
+    totalTextLength = textNodes.reduce((sum, node) => (sum += getNodeTextLength(node)), 0);
+
+    const nodesToVisit = textNodes
+      .sort((a, b) => getNodeTextLength(b) - getNodeTextLength(a))
+      .slice(0, MAX_NODES_VISITED);
+
+    const nodePromises = nodesToVisit.map(node => fetchComputedFontSize(driver, node));
+    const visitedNodes = await Promise.all(nodePromises);
+
+    /** @type {Array<NodeFontData>} */
+    const failingNodes = [];
+    for (const visitedNode of visitedNodes) {
+      if (!visitedNode) continue;
+      visitedTextLength += visitedNode.textLength;
+
+      if (visitedNode.fontSize < MINIMAL_LEGIBLE_FONT_SIZE_PX) {
+        failingNodes.push(visitedNode);
+        failingTextLength += visitedNode.textLength;
+      }
+    }
+
+    const nodesToAnalyze = failingNodes
+      .sort((a, b) => b.textLength - a.textLength)
+      .slice(0, MAX_NODES_ANALYZED);
+
+    return {totalTextLength, visitedTextLength, failingTextLength, nodesToAnalyze};
+  }
+
+  /**
+   *
+   * @param {LH.Gatherer.PassContext['driver']} driver
+   * @param {Array<NodeFontData>} nodesToAnalyze
+   */
+  static async analyzeFailingNodes(driver, nodesToAnalyze) {
```

Comment:
Maybe we could make that clearer than "analyze", since it's not really doing any analysis. `fetchSourceRule` describes itself well, so maybe something like `fetchFailingNodeSourceRules`? `sourceFailingNodes`?  :)

Reply:
done